ui/DesignSystem/Component/SegmentedControl.svelte

```diff
@@ -0,0 +1,53 @@
+<script>
+  // Currently active option.
+  export let active = null;
+  // The available options.
+  export let options = null;
+
+  const onClick = (option) => {
+    option.event();
+    currentlyActive = option.value;
+  };
+
+  $: currentlyActive = active;
+</script>
+
+<style>
+  .segmented-control {
+    display: flex;
+  }
+  .segmented-control button {
+    cursor: pointer;
+    padding: 8px 16px;
+    border-radius: 4px;
+    margin: 0;
+    background-color: var(--color-background);
+    color: var(--color-foreground-level-6);
+    font-family: var(--typeface-medium);
+  }
+  .segmented-control button:focus {
+    outline: none;
+  }
+  .segmented-control button.active {
+    background-color: var(--color-foreground-level-1);
+    color: var(--color-secondary);
+  }
+  .segmented-control button:hover {
+    background-color: var(--color-foreground-level-1);
+  }
+  .segmented-control button:active {
+    background-color: var(--color-foreground-level-1);
+    color: var(--color-foreground-level-5);
+  }
+</style>
+
+<div class="segmented-control">
+  {#each options as option}
+    <button
+      class:active={option.value === currentlyActive}
+      value={option.value}
+      on:click={() => onClick(option)}>
```

Comment:
A more Svelte-y way to do this would be to use a dispatch:

```diff
diff --git a/ui/DesignSystem/Component/SegmentedControl.svelte b/ui/DesignSystem/Component/SegmentedControl.svelte
index 5239d70..2ac79d1 100644
--- a/ui/DesignSystem/Component/SegmentedControl.svelte
+++ b/ui/DesignSystem/Component/SegmentedControl.svelte
@@ -1,11 +1,14 @@
 <script>
+  import { createEventDispatcher } from "svelte";
+  const dispatch = createEventDispatcher();
+
   // Currently active option.
   export let active = null;
   // The available options.
   export let options = null;

   const onClick = (option) => {
-    option.event();
+    dispatch("selected", option);
     currentlyActive = option.value;
   };

diff --git a/ui/Screen/DesignSystemGuide.svelte b/ui/Screen/DesignSystemGuide.svelte
index 430002c..299a7ad 100644
--- a/ui/Screen/DesignSystemGuide.svelte
+++ b/ui/Screen/DesignSystemGuide.svelte
@@ -232,17 +232,14 @@
     {
       title: "Open",
       value: 0,
-      event: () => {},
     },
     {
       title: "Closed",
       value: 1,
-      event: () => {},
     },
     {
       title: "All",
       value: 2,
-      event: () => {},
     },
   ];
 </script>
@@ -865,7 +862,12 @@
       </Swatch>

       <Swatch>
-        <SegmentedControl active={1} options={segmentedControlOptions} />
+        <SegmentedControl
+          active={1}
+          options={segmentedControlOptions}
+          on:selected={event => {
+            console.log(event.detail);
+          }} />
       </Swatch>
     </Section>
   </div>
```